packages/notebook/src/windowing.ts

```diff
@@ -235,6 +251,17 @@ export class NotebookWindowedLayout extends WindowedLayout {
       !widget.node.classList.contains(DROP_SOURCE_CLASS) &&
       widget !== this._willBeRemoved
     ) {
+      // Note: `index` is relative to the displayed cells, not all cells,
+      // hence we compare with the widget itself.
+      if (widget === this.activeCell) {
```

Comment:
The condition lines 248-252 must be changed because this will only apply to `CodeCell` currently.

Reply:
Thanks! I added a test case, and I tried moving the lines as suggested in da7e71201c569d5a8b985aa5018096db76f2c956, but moving that condition down meant that we never detach markdown/raw cells (as caught by tests) so instead I moved out the added condition from lines 256+ one level up in 15a3569684e0560dffe2c94851fddd137d247bd6.